File: solutions/nim/triangle/1/triangle.nim
proc isTriangle(sides: array[3 ,int]): bool =
  var ok = (sides[0] * sides[1] * sides[2]) > 0
  ok = ok and (sides[0] + sides[1] >= sides[2])
  ok = ok and (sides[1] + sides[2] >= sides[0])
  ok = ok and (sides[0] + sides[2] >= sides[1])
  return ok

proc isEquilateral*(sides: array[3, int]): bool =
  return isTriangle(sides) and (sides[0] == sides[1]) and (sides[0] == sides[2])

proc isIsosceles*(sides: array[3, int]): bool =
  return isTriangle(sides) and ((sides[0] == sides[1]) or (sides[1] == sides[2]) or (sides[0] == sides[2]))

proc isScalene*(sides: array[3, int]): bool =
  return isTriangle(sides) and (sides[0] != sides[1]) and (sides[1] != sides[2]) and (sides[0] != sides[2])

File: solutions/ocaml/triangle/1/triangle.ml
let is_triangle a b c =
  (a > 0) && (b > 0) && (c > 0) && (a + b >= c) && (a + c >= b) && (b + c >= a)

let is_equilateral a b c =
  (is_triangle a b c) && (a = b) && (b = c)

let is_isosceles a b c =
  (is_triangle a b c) && ((a = b) || (a = c) || (b = c))

let is_scalene a b c =
  (is_triangle a b c) && (a <> b) && (b <> c) && (a <> c)
